### ISSUE TITLE
about learning rate

### ISSUE THREAD

**Issue by kernel8liang** [bigaidream-projects/drmad#8]
Hi,

I don't understand learning rate update in optimizer ```sgd4_mad```,  the ```N_safe_sampling = len(alphas)``` decided the loop numbers .

dose ```alphas``` represent how many learning rate in the net we want to tune, for example each layer have it's own learning rate, or even each Neuron has it own learning rate.

i don't understand why ```d_alphas``` be calculated like below,
 
```
d_alphas[i] = np.dot(d_x, v)
```

**Comment by bigaidream** [comment]
As for `alphas`, suppose that we have 2 layers and train for 100 iterations, the shape of `alphas` would be [2, 100]. 

As for `d_alphas`, they are calculated based on the paper "Gradient-based Hyperparameter Optimization through Reversible Learning", Algorithm 2, Step 5.

**Comment by kernel8liang** [comment]
which experiment you used to test the ```sgd4_mad```.

**Comment by bigaidream** [comment]
https://github.com/HIPS/hypergrad/blob/master/experiments/Jan_28_training_schedule/3_nn/experiment.py

Just replace `sgd4` with `sgd4_mad`.

**Comment by kernel8liang** [comment]
thanks.
